src/apps/search-result/search-result.tsx

```diff
@@ -59,11 +60,11 @@ const SearchResult: React.FC<SearchResultProps> = ({ q, pageSize }) => {
 
   // If q changes (eg. in Storybook context)
   //  then make sure that we reset the entire result set.
-  useEffect(() => {
+  useDeepCompareEffect(() => {
     setResultItems([]);
   }, [q, pageSize, filters]);
 
-  useEffect(() => {
+  useDeepCompareEffect(() => {
```

Comment:
😮 - so there are actually already two places where this is needed.

Even better to get a fix for this in. Perhaps React Query has saved us unnecessary requests - but actually not unnecessary rerenders in other situations.